mifospay/src/main/java/org/mifos/mobilewallet/mifospay/receipt/ui/ReceiptActivity.java

```diff
@@ -195,14 +198,34 @@ public void writeReceiptToPDF(ResponseBody responseBody, String filename) {
         if (!mifosDirectory.exists()) {
             mifosDirectory.mkdirs();
         }
-        File documentFile = new File(mifosDirectory.getPath(), filename);
+        final File documentFile = new File(mifosDirectory.getPath(), filename);
         if (!FileUtils.writeInputStreamDataToFile(responseBody.byteStream(), documentFile)) {
             showToast(Constants.ERROR_DOWNLOADING_RECEIPT);
         } else {
-            showSnackbar(Constants.RECEIPT_DOWNLOADED_SUCCESSFULLY);
+            Toaster.show(findViewById(android.R.id.content),
+                    Constants.RECEIPT_DOWNLOADED_SUCCESSFULLY, Snackbar.LENGTH_LONG, Constants.VIEW,
+                    new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            openFile(ReceiptActivity.this, documentFile);
+                        }
+                    });
         }
     }
 
+    private void openFile(Context context, File file) {
+        final Uri data = FileProvider.getUriForFile(context,
+                "org.mifos.mobilewallet.mifospay.provider", file);
+        context.grantUriPermission(context.getPackageName(),
+                data, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        Intent intent = new Intent(Intent.ACTION_VIEW)
+                .setDataAndType(data, "application/pdf")
+                .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+        intent = Intent.createChooser(intent, "View Receipt");
```

Comment:
`View Receipt` is a UI related text and it needs to be translated, so please add it in `string.xml` file

Reply:
@dubeyaayush07 Please look into this.

Reply:
@luckyman20 Done